game.py

```diff
@@ -1,1091 +1,101 @@
-import datetime
-import json
-import os
-import random
 import time
-import typing
-from itertools import chain
 
 import pyxel
 
-from board import Board
-from calculator import clamp, attack, get_setl_totals, complete_construction, attack_setl
-from catalogue import get_available_improvements, get_available_blessings, get_available_unit_plans, get_heathen, \
-    get_default_unit, get_improvement, get_blessing, get_unit_plan, Namer, FACTION_COLOURS, PROJECTS, get_project
-from menu import Menu, MainMenuOption, SetupOption, WikiOption
-from models import Player, Settlement, Construction, OngoingBlessing, CompletedConstruction, Unit, HarvestStatus, \
-    EconomicStatus, Heathen, AttackPlaystyle, GameConfig, Biome, Victory, VictoryType, AIPlaystyle, \
-    ExpansionPlaystyle, UnitPlan, OverlayType, Faction, ConstructionMenu, Project
-from movemaker import MoveMaker, set_player_construction
-from music_player import MusicPlayer
-from overlay import SettlementAttackType, PauseOption
-from save_encoder import SaveEncoder, ObjectConverter
-
-# The prefix attached to save files created by the autosave feature.
-AUTOSAVE_PREFIX = "auto"
-# The directory where save files are created and loaded from.
-SAVES_DIR = "saves"
+from game_management.game_controller import GameController
+from game_management.game_input_handler import on_key_arrow_down, on_key_arrow_up, on_key_arrow_left, \
+    on_key_arrow_right, on_key_return, on_mouse_button_right, on_mouse_button_left, on_key_shift, on_key_c, on_key_f, \
+    on_key_d, on_key_tab, on_key_space, on_key_m, on_key_s, on_key_n, on_key_b, on_key_escape, on_key_a, on_key_j
+from game_management.game_state import GameState
 
 
 class Game:
     """
-    The main class for the game. Contains the majority of business logic, and none of the drawing.
+    The main class for the game. Contains the main game loop and subsequent state and controls, and none of the drawing.
+    The state of the game is available in GameState - and game controlling mechanisms are available in GameController.
     """
 
     def __init__(self):
         """
         Initialises the game.
         """
-        pyxel.init(200, 200, title="Microcosm", display_scale=5, quit_key=pyxel.KEY_NONE)
-
-        self.menu = Menu()
-        self.board: typing.Optional[Board] = None
-        self.players: typing.List[Player] = []
-        self.heathens: typing.List[Heathen] = []
-
-        self.on_menu = True
-        self.game_started = False
-
-        self.last_time = time.time()
-
-        # The map begins at a random position.
-        self.map_pos: (int, int) = random.randint(0, 76), random.randint(0, 68)
-        self.turn = 1
+        pyxel.init(200, 200, title="Microcosm", quit_key=pyxel.KEY_NONE)
```

Comment:
Looks like we lost the `display_scale=5` in the conflict resolution

Reply:
[588b895](https://github.com/ChrisNeedham24/microcosm/pull/111/commits/588b895112c545f17b9883b616153d3514b540f7)